src/main/scala/scalapb/json4s/JsonFormat.scala

```diff
@@ -384,21 +388,26 @@ class Parser private (config: Parser.ParserConfig) {
     }
   }
 
-  def defaultEnumParser(enumDescriptor: EnumDescriptor, value: JValue): EnumValueDescriptor = value match {
-    case JInt(v) => enumDescriptor.findValueByNumber(v.toInt).getOrElse(throw new JsonFormatException(s"Invalid enum value: ${v.toInt} for enum type: ${enumDescriptor.fullName}"))
-    case JString(s) =>
-      enumDescriptor.values.find(_.name == s).getOrElse(throw new JsonFormatException(s"Unrecognized enum value '${s}'"))
-    case _ =>
-      throw new JsonFormatException(
-        s"Unexpected value ($value) for enum ${enumDescriptor.fullName}")
-  }
+  def defaultEnumParser(enumDescriptor: EnumDescriptor, value: JValue): Option[EnumValueDescriptor] = value match {
+      case JInt(v) => enumDescriptor.findValueByNumber(v.toInt)
+          .orElse(Some(enumDescriptor.findValueByNumberCreatingIfUnknown(v.toInt)))
+      case JString(s) if config.isIgnoringUnknownFields => enumDescriptor.values.find(_.name == s)
+          .orElse(Some(NullValue.NULL_VALUE.scalaValueDescriptor))
```

Comment:
this needs to throw for java compat?

Reply:
You are right. Sorry - did all changes for this one and then left it out:/

Reply:
Ups. Did all the changes for this one and left if `git stash`ed somewhere. Sorry. Fixed.